testng-core-api/src/main/java/org/testng/ITestClassInstance.java

```diff
@@ -0,0 +1,36 @@
+package org.testng;
+
+/** Represents the ability to retrieve the parameters associated with a factory method. */
+public interface ITestClassInstance {
+
+  /** @return - The actual instance associated with a factory method */
+  Object getInstance();
+
+  /**
+   * @return - The actual index of instance associated with a factory method. This index has a 1:1
+   *     correspondence with what were specified via the <code>indices</code> attribute of the
+   *     <code>@Factory</code> annotation. For e.g., lets say you specified the indices to the "1"
+   *     and your factory returned 4 instances, then the instance on which this method is invoked
+   *     would have the value as "1".
+   */
+  int getIndex();
+
+  /**
+   * @return - returns an index which indicates the running position in the array of test class
+   *     instances that were produced by a <code>@Factory</code> annotated constructor or static
+   *     method. For e.g., lets say your <code>@Factory</code> method returned 4 instances, then
+   *     each of the invocations to this method would return a value from <code>0</code> to <code>3
+   *     </code>
+   */
+  int getInvocationIndex();
+
+  /** @return - The parameters associated with the factory method as an array. */
+  Object[] getParameters();
```

Comment:
Could you try to remove it from the interface?

Reply:
If I remove this from this interface then we would basically loose track as to what all parameters were used to create the instance (which is what the user would want to know).

Reply:
As I understand from the code, they are not params of the instance but params of the factory method.

Reply:
3. `getFactoryMethod().getParameters()`

Reply:
What should `getFactoryMethod()` return? Like I mentioned before, it is NOT being stored anywhere. It gets created on the fly, and then it gets discarded

Reply:
It could be an `IFactoryMethod` which is currently missing from the api.
We can start by a first method.